### ISSUE TITLE
'View as HTML' label is meaningless to the citizen

### ISSUE THREAD

**Issue by equivalentideas** [mysociety/alaveteli#2303]
For the link citizens use to view a document attached to a request in their browser:

![screen shot 2015-04-07 at 12 16 03 pm](https://cloud.githubusercontent.com/assets/1239550/7015983/1630e22a-dd20-11e4-87fc-5bceeee014db.png)

Why are we telling them what format it's in? Do they know what it means that it's in HTML?

Maybe a label like: 'View online' or 'View in your browser' or 'View text version' would be better?

I guess the key questions is 'what is the benefit of viewing it in HTML in your browser?' We should try and make the action label focus on the use to the citizen.

On the document view page we also say "This is an HTML version of an attachment to the Freedom of Information request". Same question applies here.

**Comment by garethrees** [comment]
I quite like "View in your browser"

**Comment by crowbot** [comment]
I think one of the main reasons for having the HTML view originally was to allow google to index the contents of the documents, so I guess the users benfitting were the ones who found what they were looking for from a google search.

**Comment by garethrees** [comment]
> I think one of the main reasons for having the HTML view originally was to allow google to index the contents of the documents, so I guess the users benfitting were the ones who found what they were looking for from a google search.

As I understand it, I think the issue is more with the wording of the link rather than the actual use of viewing in the browser.

We could just have "Download" and "View"?

**Comment by crowbot** [comment]
On 14 April 2015 at 11:05, Gareth Rees <notifications@github.com> wrote:

> I think one of the main reasons for having the HTML view originally was to
> allow google to index the contents of the documents, so I guess the users
> benfitting were the ones who found what they were looking for from a google
> search.
>
> As I understand it, I think the issue is more with the wording of the link
> rather than the actual use of viewing in the browser.
>
Yes, Luke's question was about what the benefit was to the user of having
the link - so that we can give the link a name that reflects that benefit.
My point was that the main benefit is rather indirect.

We could just have "Download" and "View"?
>
We could, but depending on people's setups, the "Download" link may also
display the document in the browser.

**Comment by garethrees** [comment]
> We could, but depending on people's setups, the "Download" link may also
display the document in the browser.

Yeah that's true, and could be confusing.

**Comment by equivalentideas** [comment]
We had a little exchange on the use of the doc web view in https://github.com/openaustralia/righttoknow/issues/473 :

> It would be interesting to see if there are any documented use cases, but here's an idea:

> * As a user I need to view request documents without downloading them onto my device, which does not have this capability

> I'm not quite sure what processing this does to the pdf text, but it may been making it more accessible to assistive technologies in the browser than the pdf would be otherwise. It might be more digestible to search engines as well, depending on the markup and meta tags etc.

> It gives us the opportunity to add information about the document in the header, such as linking back to the request, to Right To Know, etc. I think that's quite important.

Thinking about it more, and chatting with @henare , I think that last point is particularly important. It's a great page to share because it provides the reference back to the request and gives context to the document. We could actually add some nice share buttons onto it to see if that encourages more sharing. #2285 is related somehow :)

I think "View in your browser", or "View on the web" are a positive iteration from "View as HTML", because at least it's describing the action/context rather than the format.

**Comment by garethrees** [comment]
I'm totally convinced. It does seem like a page we can make more use of.

> We could, but depending on people's setups, the "Download" link may also
> display the document in the browser.

What about `Download | View on {{ SITE_NAME }}"?

I think most people will be used to their browser behaviour, so when they click "Download" on a PDF they're probably going to expect it to just open in the browser if they've ever opened a PDF in their browser before. "View on WDTK" avoids any "technical" terms and is quite accurate in that we're rendering the document in the context of the website?

**Comment by equivalentideas** [comment]
Nice one @garethrees , I think "View on {{ SITE_NAME }}" makes sense. Could that be an issue with some of the longer names?

**Comment by garethrees** [comment]
Looks okay to me

![screen shot 2015-04-15 at 12 37 07](https://cloud.githubusercontent.com/assets/282788/7157934/2b053032-e36c-11e4-8185-09823495d7f0.png)

**Comment by crowbot** [comment]
Neither of these are really accurate labels though, because 'download' won't always download, and both are coming from the site. 

How about as an alternative "Original attachment" and "View as web page"

**Comment by garethrees** [comment]
> How about as an alternative "Original attachment" and "View as web page"

Those work too.

@zarino @wrightmartin team design, do you have an existing pattern for this?

**Comment by zarino** [comment]
Just trying to think of examples of this sort of thing elsewhere on my computer…

In Gmail, the default behaviour from clicking an attachment is to render it in the browser, no matter what it is. Then hovering over the attachment icon, you get options to download it to your device, or add it to cloud storage:

![google-drive](https://cloud.githubusercontent.com/assets/739624/7177950/e0c29d68-e422-11e4-8109-e286f29c886c.gif)

On a Mac, you can open a file (in the default app, or a different one) or you can "QuickLook" the file in the Finder, which is the desktop equivalent of rendering a file in the browser rather than downloading it:

![mac](https://cloud.githubusercontent.com/assets/739624/7177968/fac96700-e422-11e4-87a4-bbddb1748280.gif)

In Slack, the two options are "Open original" (which opens the file in your web browser) and "Download":

![slack](https://cloud.githubusercontent.com/assets/739624/7178126/4ef49466-e424-11e4-9a6a-b716381043e0.gif)

I wonder whether "Open" and "Download" should be the two options?

(And to answer @crowbot's point, I wouldn't be too worried about "Download" not technically downloading the file in all cases. Unless you've got some user testing that shows this is a point of confusion for Alaveteli users?)

**Comment by crowbot** [comment]
@zarino - where "Open" is the HTML version and "Download" the PDF version? If so, I think we might need to tweak it to account for the fact that HTML conversion sometimes mangles the output - we should perhaps indicate that the PDF is the original.

**Comment by zarino** [comment]
> where "Open" is the HTML version and "Download" the PDF version?

Yes.

In the case where "Open" shows an automatically-converted HTML representation of the file, I'd expect a nice clear "Download" (maybe "Download original"?) button on the same screen, so if it looks broken, I can try downloading instead.

**Comment by crowbot** [comment]
Yeah, I think we do that reasonably well at the moment 
![screen shot 2015-04-16 at 11 32 45](https://cloud.githubusercontent.com/assets/2912/7179093/57c46cee-e42c-11e4-936b-9abc5f404fd9.png)

**Comment by wrightmartin** [comment]
Thanks @zarino, there is nothing for m to add here.

**Comment by equivalentideas** [comment]
> where "Open" is the HTML version and "Download" the PDF version?

:+1:

**Comment by garethrees** [comment]
Basecamp have a nice pattern for something similar:

<img width="447" alt="screen shot 2017-01-23 at 22 40 20" src="https://cloud.githubusercontent.com/assets/282788/22226139/0d894538-e1bd-11e6-9243-26dcef3eb5ba.png">

- View full size: Open in browser
- Download: Forces a download

**Comment by equivalentideas** [comment]
Looking at this again, I think "Download original" captures that concept well.

What about "Quick view" instead of "open"? Looking at the examples @zarino [posted above](https://github.com/mysociety/alaveteli/issues/2303#issuecomment-93692147), that seems to capture the idea that you're glancing at something, and that it [may not render perfectly](https://github.com/mysociety/alaveteli/issues/2303#issuecomment-93701938).

**Comment by RichardTaylor** [comment]
A user recently wrote to WhatDoTheyKnow saying they couldn't find their response. 

They were happy when a direct link to the PDF released was provided to them by email, along with a statement:

>To read the response you need to click on either "Download" or "View as HTML" next to the attachment which is shown at the top of the message.

**Comment by garethrees** [comment]
[manualslib](https://www.manualslib.com/) uses "Download PDF" which forces a download (rather than open in browser), and "View in Browser" (which renders the raw PDF in browser). Their "default" view is to render the the file in a custom file viewer.

DEFAULT:

![Screenshot 2023-10-14 at 11 58 45](https://github.com/mysociety/alaveteli/assets/282788/b1376e58-8986-462c-9484-08fc5bbe5764)


OTHER OPTIONS:

![Screenshot 2023-10-14 at 11 57 03](https://github.com/mysociety/alaveteli/assets/282788/d81b8701-26dd-41e8-96af-a132c0705e4f)

**Comment by HelenWDTK** [comment]
This issue has been automatically closed due to a lack of discussion or resolution for over 12 months.
Should we decide to revisit this issue in the future, it can be reopened.
